jb/child_allowance.py

```diff
@@ -0,0 +1,236 @@
+# Roadmap:
+# Comparison to other potential policies
+# Replicate plotly code from the child allowance for JB
+# Move to second simulation
+
+# Preamble and read data
+import microdf as mdf
+import pandas as pd
+import numpy as np
+import us
+
+# Read in census data and specify columns for use
+raw = pd.read_csv(
+    "https://github.com/UBICenter/child-allowance/blob/master/jb/data/cps_00003.csv.gz",
+    usecols=[
+        "YEAR",
+        "MONTH",
+        "STATEFIP",
+        "AGE",
+        "SEX",
+        "RACE",
+        "HISPAN",
+        "SPMWT",
+        "SPMFTOTVAL",
+        "SPMTOTRES",
+        "SPMCHXPNS",
+        "SPMTHRESH",
+        "SPMFAMUNIT",
+        "ASECWT",
+    ],
+)
+
+# Create a copy of the raw dataset and make column names non-capitalized
+# for readability
+person = raw.copy(deep=True)
+person = person[~person.ASECWT.isnull()]
+person.columns = person.columns.str.lower()
+
+# Asec weights are year-person units, and we average over 3 years,
+# so divide by the 3 years to give per year weights.
+person.asecwt /= 3
+
+# Define child age identifiers
+person["child_6"] = person.age < 6
+person["infant"] = person.age < 1
+person["toddler"] = person.age.between(1, 2)
+person["preschool"] = person.age.between(3, 5)
+person["person"] = 1
+
+# Redefine race categories
+person.race.mask(
+    person.race.between(200, 999, inclusive=False), "Other", inplace=True
+)
+person.race.mask(person.race == 100, "White", inplace=True)
+person.race.mask(person.race == 200, "Black", inplace=True)
+person.race.mask(person.race == 999, "Race unknown", inplace=True)
+
+# Redefine hispanic categories
+person.hispan.mask(
+    person.hispan.between(100, 612, inclusive=True), "Hispanic", inplace=True
+)
+person.hispan.mask(person.hispan == 0, "Not Hispanic", inplace=True)
+person.hispan.mask(
+    (person.hispan != "Hispanic") & (person.hispan != "Not Hispanic"),
+    "Hispanic status unknown",
+    inplace=True,
+)
+
+# Combine race + hispanic categories
+person["race_hispan"] = "Other or Unknown"
+person.race_hispan.mask(
+    (person.race == "White") & (person.hispan == "Not Hispanic"),
+    "White non-Hispanic",
+    inplace=True,
+)
+person.race_hispan.mask(
+    (person.race == "Black") & (person.hispan == "Not Hispanic"),
+    "Black",
+    inplace=True,
+)
+person.race_hispan.mask(
+    (person.race == "Other") & (person.hispan == "Not Hispanic"),
+    "Other non-Hispanic",
+    inplace=True,
+)
+person.race_hispan.mask(person.hispan == "Hispanic", "Hispanic", inplace=True)
+
+# Relabel sex categories
+person["sex"].mask(person["sex"] == 1, "Male", inplace=True)
+person["sex"].mask(person["sex"] == 2, "Female", inplace=True)
+
+# Create State categories
+person["state"] = (
+    pd.Series(person.statefip)
+    .apply(lambda x: us.states.lookup(str(x).zfill(2)).name)
+    .tolist()
+)
+
+# Define data collected at the SPM unit level
+spmu_cols = [
```

Comment:
Make `SPMU_COLS` since it's a constant

Reply:
As in just capitalize all instances of SPMU_COLS?